lib/DDG/Spice/Twitter.pm

```diff
@@ -5,7 +5,12 @@ use strict;
 use DDG::Spice;
 
 spice to => 'https://duckduckgo.com/tw.js?user=$1&callback={{callback}}&current=1';
-triggers query => qr/^(?:twitter\s)?@([a-z0-9_]+)$|^twitter\s([a-z0-9_]+)$/i;
+spice proxy_cache_valid => '200 1h';
+
+my $grammar = qr/(?:on)?(?:at)?/io;
+my $other = qr/(?:user)?(?:account)?(?:id)?/io;
+
+triggers query => qr/^(?:twitter\s)?(?:$other\s)?@([a-z0-9_]+)$|^twitter\s(?:$other\s)?([a-z0-9_]+)$|^([a-z0-9_]+)\s(?:$other\s)?(?:$grammar\s)?(?:twitter)$|^@([a-z0-9_]+)\s(?:$other\s)?(?:$grammar\s)?(?:twitter)?$/i;
```

Comment:
Let's split this up to make it more legible. We can use multiple `triggers`. Also we might as well use `query_lc` so the regex doesn't have to be case sensitive.

```perl
triggers query_lc =>^(?:twitter\s)?@([a-z0-9_]+)$;
triggers query_lc =>^twitter\s([a-z0-9_]+)$;
...
triggers query_lc =>^...$;
```